dask/dataframe/io/io.py

```diff
@@ -750,5 +755,245 @@ def sorted_division_locations(seq, npartitions=None, chunksize=None):
     return values, positions
 
 
+class _PackedArgCallable(DataFrameIOFunction):
+    """Packed-argument wrapper for DataFrameIOFunction
+
+    This is a private helper class for ``from_map``. This class
+    ensures that packed positional arguments will be expanded
+    before the underlying function (``func``) is called. This class
+    also handles optional metadata enforcement and column projection
+    (when ``func`` satisfies the ``DataFrameIOFunction`` protocol).
+    """
+
+    def __init__(
+        self,
+        func,
+        args=None,
+        kwargs=None,
+        meta=None,
+        packed=False,
+        enforce_metadata=False,
+    ):
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+        self.meta = meta
+        self.enforce_metadata = enforce_metadata
+        self.packed = packed
+        self.is_dataframe_io_func = isinstance(self.func, DataFrameIOFunction)
```

Comment:
If possible it seems like it would be better not to use `isinstance` here but instead rely on duck typing (or protocol matching I guess as @ian-r-rose suggested).

Reply:
Right - My goal here is to check that `self.func` satisfies the `DataFrameIOFunction` protocol. If there is a more canonical way to do it, I'll be happy to change this (I'm still pretty novice on python protocols)

Reply:
nice!